src/dodal/devices/aperturescatterguard.py

```diff
@@ -83,24 +80,33 @@ def as_list(self) -> List[SingleAperturePosition]:
         ]
 
 
-class ApertureScatterguard(InfoLoggingDevice):
-    aperture = Cpt(Aperture, "-MO-MAPT-01:")
-    scatterguard = Cpt(Scatterguard, "-MO-SCAT-01:")
-    aperture_positions: AperturePositions | None = None
-    TOLERANCE_STEPS = 3  # Number of MRES steps
+class ApertureScatterguard(StandardReadable):
+    def __init__(self, prefix: str = "", name: str = "") -> None:
+        self.aperture = Aperture(prefix="-MO-MAPT-01:")
+        self.scatterguard = Scatterguard(prefix="-MO-SCAT-01:")
+        self.aperture_positions: AperturePositions | None = None
+        self.TOLERANCE_STEPS = 3  # Number of MRES steps
+        self.selected_aperture = self.SelectedAperture(
+            backend=SimSignalBackend(datatype=SingleAperturePosition, source="")
+        )
+        self.set_readable_signals(
+            read=[
+                self.selected_aperture,
+            ]
+        )
+        super().__init__(name)
 
-    class SelectedAperture(SignalRO):
-        def get(self):
+    class SelectedAperture(SignalR):
+        async def read(self, *args, **kwargs):
             assert isinstance(self.parent, ApertureScatterguard)
-            return self.parent._get_current_aperture_position()
-
-    selected_aperture = Cpt(SelectedAperture)
+            await self._backend.put(await self.parent._get_current_aperture_position())
+            return {self.name: await self._backend.get_reading()}
 
     def load_aperture_positions(self, positions: AperturePositions):
         LOGGER.info(f"{self.name} loaded in {positions}")
         self.aperture_positions = positions
 
-    def set(self, pos: SingleAperturePosition) -> StatusBase:
+    def set(self, pos: SingleAperturePosition):
```

Comment:
set does need to return an `AsyncStatus` in the end, so that the Bluesky RunEngine knows what to do with it